packages/hydrogen/src/customer/customer.ts

```diff
@@ -128,16 +143,50 @@ type CustomerClientOptions = {
   waitUntil?: ExecutionContext['waitUntil'];
   /** This is the route in your app that authorizes the user after logging in. Make sure to call `customer.authorize()` within the loader on this route. It defaults to `/account/authorize`. */
   authUrl?: string;
+  /** This is the route in your app that we redirect to whenever query/mutate was called with a not login user. Make sure to call `customer.login()` within the loader on this route. It defaults to `/account/login`. */
+  loginUrl?: string;
+  /** The behaviour when query/mutate is called without a LoggedIn customer. The default behaviour is redirecting to /account/login where login() is located and return to the current location after auth. */
+  unauthorizedHandler?: () => DataFunctionValue;
 };
 
+const DEFAULT_LOGIN_URL = '/account/login';
+const DEFAULT_AUTH_URL = '/account/authorize';
+const DEFAULT_REDIRECT_PATH = '/account';
+
+function defaultUnauthorizedHandler(request: CrossRuntimeRequest) {
+  const redirectPath = request.url ? new URL(request.url).pathname : undefined;
+
+  const loginUrlWithRedirectPath =
+    DEFAULT_LOGIN_URL +
+    (redirectPath
+      ? `?${new URLSearchParams(`redirectPath=${redirectPath}`).toString()}`
```

Comment:
this redirect path needs to be protected from open redirect vulnerability
ex: https://github.com/Shopify/hydrogen/pull/1629/files#diff-8fdaae27e8801720b17428cfac5986e638c1323bdfbb87072513cbcbae679a3bR77

Reply:
I think for consistency, we should use the `return_to` and/or `redirect` search params instead: https://github.com/Shopify/hydrogen/blob/main/packages/hydrogen/src/routing/redirect.ts#L56